### ISSUE TITLE
Deploying a VPN Gateway with sku=basic fails when enable_bgp=false

### ISSUE THREAD

**Issue by sjovang** [Azure/terraform-azurerm-caf-enterprise-scale#369]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.1.9

**azure provider**: 3.5.0

**module**: 2.0.1

### Description

Deploying a VPN Gateway with sku=basic fails  with the error message:

```
Error: Creating/Updating Virtual Network Gateway: (Name "Myorg-vpngw-norwayeast" / Resource Group "Myorg-connectivity-norwayeast"): network.VirtualNetworkGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BasicSkuVirtualNetworkGatewayDoesNotSupportBgp" Message="A virtual network gateway SKU of Standard or higher is required for BGP support on virtual network gateway /subscriptions/<subscription id redacted>/resourceGroups/Capitech-connectivity-norwayeast/providers/Microsoft.Network/virtualNetworkGateways/Myorg-vpngw-norwayeast" Details=[]
```

1. Configure settings.connectivity.tf with virtual_network_gateway enabled, gateway_sku_vpn=Basic and enable_bgp=false:

```
virtual_network_gateway = {
              enabled = true
              config = {
                address_prefix           = "10.11.12.0/26"
                gateway_sku_expressroute = ""
                gateway_sku_vpn          = "Basic"
                advanced_vpn_settings = {
                  enable_bgp                       = false
                  active_active                    = false
                  private_ip_address_allocation    = ""
                  default_local_network_gateway_id = ""
                  vpn_client_configuration         = []
                  bgp_settings                     = []
                  custom_route                     = []
                }
              }
            }
```

2. run terraform apply
Same error message is displayed when setting enable_bgp=null

**Comment by krowlandson** [comment]
Thank you for logging this issue @sjovang

It may be that we need to allow a value of `null` on this setting instead of always setting to `true` or `false` using a `coalesce()` function.

https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/9a1b66e613dd363dab29995879b9dc0764bf07c7/modules/connectivity/locals.tf#L660

We will test and come back to you.

cc: @matt-FFFFFF

**Comment by jtracey93** [comment]
Hey @sjovang,

The `basic` SKU for VPN GWs doesn't not support BGP as documented here: https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-bgp-overview#is-bgp-supported-on-all-azure-vpn-gateway-skus

I would also advise against using the `basic` SKU for VPN GWs as this is a legacy SKU and has feature limitations etc. as documented here https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-about-skus-legacy and therefore would suggest `VpnGw1` or `VpnGw2` and preferably `generation 2` as these have higher performance throughputs.

Thanks

**Comment by sjovang** [comment]
> 

I agree that we, in general should avoid using the `basic` sku for the reasons you mention above, but there also exist use cases where the `basic` sku solves the problem without adding the extra cost of more feature rich gateways.

Maybe it is a small environment where 100mbps throughput is more than sufficient, maybe the client is very cost sensitive, maybe you only need the virtual network gateway for transitive vnet peering etc.

**Comment by matt-FFFFFF** [comment]
Hi @sjovang 

I have tested a fix, visible in #376

I'll add this to the next patch release, v2.0.2
